src/microbe_stage/MicrobeSpecies.cs

```diff
@@ -121,4 +121,39 @@ public override object Clone()
 
         return result;
     }
+
+    public float BaseOsmoregulationCost()
+    {
+        return Organelles.Sum(organelle => organelle.Definition.HexCount)
+            * MembraneType.OsmoregulationFactor
+            * Constants.ATP_COST_FOR_OSMOREGULATION;
```

Comment:
If I recall well, the editor uses a pretty similar method to compute osmoregulation, and thus it might be good to unify this in case further factors are added.

Reply:
I went ahead and refactored the crap out of this area of the code; feel free to re-evaluate.